File: apps/silver-oak-ui/src/app/view/characters/model/CharactersResponseDTO.ts
export class CharactersResponseDTO {
  constructor(
    public id: number,
    public maxLife: number,
    public life: number,
    public damage: number,
    public level: number,
    public experience: number,
    public characterClass: Classes
  ) {}
}

class Classes {
  constructor(
    public usableWeapons: string[]
  ) {}
}

File: apps/silver-oak-core/src/main/java/com/api/silver_oak_core/model/charaters/CharactersResponseDTO.java
package com.api.silver_oak_core.model.charaters;

import com.api.silver_oak_core.model.classes.Classes;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

@Data
@AllArgsConstructor
@Builder
public class CharactersResponseDTO {
  private long id;

  private int maxLife;

  private int life;

  private int damage;

  private int level;

  private int experience;

  private Classes characterClass;
}
